lxd/device/nic_routed.go

```diff
@@ -205,6 +206,43 @@ func (d *nicRouted) validateEnvironment() error {
 	return nil
 }
 
+func (d *nicRouted) checkIPAvailability(parent string) error {
+	var addresses []net.IP
+	ipv4AddrStr, ok := d.config["ipv4.address"]
+	if ok {
+		ipv4Addrs := util.SplitNTrimSpace(ipv4AddrStr, ",", -1, true)
+		for _, addr := range ipv4Addrs {
+			addresses = append(addresses, net.ParseIP(addr))
+		}
+	}
+
+	ipv6AddrStr, ok := d.config["ipv6.address"]
+	if ok {
+		ipv6Addrs := util.SplitNTrimSpace(ipv6AddrStr, ",", -1, true)
+		for _, addr := range ipv6Addrs {
+			addresses = append(addresses, net.ParseIP(addr))
+		}
+	}
+
+	errs := make(chan error)
+	for _, address := range addresses {
+		go func(address net.IP) {
+			ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+			defer cancel()
+			errs <- isIPAvailable(ctx, address, parent)
+		}(address)
+	}
+
+	var err error
+	for range addresses {
+		if e := <-errs; e != nil {
+			err = e
+		}
+	}
```

Comment:
With the change suggested above, this can go back to what you had before:
```
for i := range(len(addresses)) {
    err := <errs
    if err != nil {
         return err
    }
}
```